Connect/Tests/Connect.Tests.ps1

```diff
@@ -0,0 +1,122 @@
+        }
+    
+        It 'Get-AzureStackNatServerAddress should be exported' {
+            Get-Command -Name Get-AzureStackNatServerAddress -ErrorAction SilentlyContinue | 
+                Should Not Be $null
+        }
+    
+        It 'Add-AzureStackVpnConnection should be exported' {
+            Get-Command -Name Add-AzureStackVpnConnection -ErrorAction SilentlyContinue | 
+                Should Not Be $null
+        }
+    
+        It 'Connect-AzureStackVpn should be exported' {
+            Get-Command -Name Connect-AzureStackVpn -ErrorAction SilentlyContinue | 
+                Should Not Be $null
+        }
+    }
+}
+
+InModuleScope $script:ModuleName {
+
+    
+    $HostComputer = $global:HostComputer
+    $Domain = $global:Domain 
+    $natServer = $global:natServer 
+    $AdminUser= $global:AdminUser 
+    $AadServiceAdmin = $global:AadServiceAdmin 
+
+    $AdminPassword = $global:AdminPassword
+    $AadServiceAdminPassword = $global:AadServiceAdminPassword
+    $stackLoginCreds = $global:AzureStackLoginCredentials
+
+    $VPNConnectionName = $global:VPNConnectionName
+
+    Set-Item wsman:\localhost\Client\TrustedHosts -Value $HostComputer -Concatenate
+    Set-Item wsman:\localhost\Client\TrustedHosts -Value mas-ca01.azurestack.local -Concatenate
+
+    Describe 'ConnectModule - Accessing Environment Data' {
+        It 'Recovered AAD Tenant ID should be correct' {
+            $global:AadTenantID = Get-AzureStackAadTenant  -HostComputer $HostComputer -Domain $Domain -User $AdminUser -Password $AdminPassword 
+            Write-Verbose "Aad Tenant ID is $global:AadTenantID" -Verbose
+            $global:AadTenantID | Should Not Be $null
+        }
+
+        It 'Get-AzureStackNatServerAddress should return valid NAT address' {
+            $script:NatIPAddress = Get-AzureStackNatServerAddress -natServer $natServer -HostComputer $HostComputer -Domain $Domain -User $AdminUser -Password $AdminPassword 
+            Write-Verbose "Returned NAT IP Address of $natIPAddress" -Verbose
+            [IPAddress]$script:NatIPAddress | Should Not Be $null
+        }
+
+        It 'Add-AzureStackVpnConnection should correctly return a VPN connection to a One Node' {
+            Add-AzureStackVpnConnection -ServerAddress $script:NatIPAddress -ConnectionName $VPNConnectionName -Password $AdminPassword
+            Get-VpnConnection -Name $VPNConnectionName | Should Not Be $null
+        }
+
+        It 'Connect-AzureStackVpn should successfully connect to a One Node environment' {
+            {Connect-AzureStackVpn -ConnectionName $VPNConnectionName -User $AdminUser -Domain $Domain -Password $AdminPassword} | Should Not Throw
+        }
+
+        It 'Add-AzureStackAzureRmEnvironment should successfully add a One Node environment' {
+            Remove-AzureRmEnvironment -Name "AzureStack" -ErrorAction SilentlyContinue -Force
+            Add-AzureStackAzureRmEnvironment -AadTenant $global:AadTenantID -Domain $Domain
```

Comment:
No changes needed immediately, I think we should start taking ARM endpoint here instead of $Domain, (with the admin ARM/User ARM split, there will be two endpoints depending on what user wants to talk to)